### PR TITLE
CI build rework

### PR DESCRIPTION
**Pull request by Oscar-Pepper**

### CODE REVIEW

**Review by fluidvanadium — COMMENTED** [review]
why rename this file? it will make all my merging across these wonderful actions harder

**Comment by Oscar-Pepper** [comment]
I wanted to match with zingolib, is it a pain to rebase your PR ontop of
the name change?

**Comment by fluidvanadium** [comment]
> 

it *might* be, and i just read that .yaml is the recommended extension.

**Comment by Oscar-Pepper** [comment]
> > 
> 
> it _might_ be, and i just read that .yaml is the recommended extension.

ok no probs i've changed it to .yaml

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -3,74 +3,152 @@ name: CI
...
-      - name:                Build NDK
-        run:                 cd rust && ./build.sh
-      - name:                Upload native rust
-        uses:                actions/upload-artifact@v3
-        with:
-          name:              zingomobile-jniLibs-${{ steps.create-timestamp.outputs.timestamp }}
-          path:              android/app/src/main/jniLibs/
+      - name: Create timestamp
+        id: create-timestamp
+        run: echo "timestamp=$(date +%Y%m%d%H%M%S)" >> $GITHUB_OUTPUT
```
Comment (fluidvanadium) [inline .github/workflows/ci.yaml:27]: im not familiar with this syntax

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -3,74 +3,152 @@ name: CI
...
 
-  test:
-    name:                    Test
-    runs-on:                 ubuntu-22.04
+  build:
+    name: Build native rust (${{ matrix.arch }})
+    runs-on: ubuntu-22.04
+    strategy:
+      matrix:
+        include:
```
Comment (fluidvanadium) [inline .github/workflows/ci.yaml:34]: strange that this matrix is different from the android targets matrix

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -3,74 +3,152 @@ name: CI
...
+      - name: Run integration tests
+        uses: reactivecircus/android-emulator-runner@v2
+        with:
+          working-directory: ./android
+          api-level: ${{ matrix.api-level }}
+          arch: ${{ matrix.arch }}
+          force-avd-creation: false
+          emulator-options: -no-snapshot-save -no-window -gpu swiftshader_indirect -noaudio -no-boot-anim -camera-back none
+          disable-animations: true
+          script: ./gradlew connectedCheck -Pandroid.testInstrumentationRunnerArguments.class=org.ZingoLabs.Zingo.IntegrationTestSuite
```
Comment (fluidvanadium) [inline .github/workflows/ci.yaml:154]: amazing!

**Review by fluidvanadium — DISMISSED** [review]

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (Oscar-Pepper) [reply .github/workflows/ci.yaml:27]: echo "OUTPUT_NAME=$(COMMAND) >> $GITHUB_OUTPUT
is standard way to store an output of a job

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (Oscar-Pepper) [reply .github/workflows/ci.yaml:34]: sorry which is the android targets matrix?

**Review by zancas — APPROVED** [review]
LGTM

**Comment by zancas** [comment]
Is this blocked on me?  What's up?

**Pull request merged**